diff --git a/BUILD b/BUILD
index d0048cb..0bf3b0f 100644
--- a/BUILD
+++ b/BUILD
@@ -14,3 +14,11 @@ genrule(
     stamp = 1,
     tools = [":print_version_header.sh"],
 )
+
+platform(
+    name = "izumi_gbmc",
+    constraint_values = [
+        "@platforms//os:linux",
+        "@platforms//cpu:armv7",
+    ],
+)
diff --git a/MODULE.bazel b/MODULE.bazel
index f5140ad..fc84c8c 100644
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -18,5 +18,10 @@
     shallow_since = "1649581036 +0200",
 )
 
+bazel_dep(name = "platforms", version = "0.0.10")
+register_toolchains(
+    "//toolchain:cc_toolchain_for_linux_arm"
+)
+
 bazel_dep(name = "googletest", version = "1.15.2")
 bazel_dep(name = "rules_cc", version = "0.1.4")
diff --git a/toolchain/BUILD b/toolchain/BUILD
new file mode 100644
index 0000000..4bc011e
--- /dev/null
+++ b/toolchain/BUILD
@@ -0,0 +1,34 @@
+package(default_visibility = ["//visibility:public"])
+load(":cc_toolchain_config.bzl", "cc_toolchain_config")
+
+filegroup(name = "empty")
+cc_toolchain_config(name = "linux_arm_toolchain_config")
+
+
+cc_toolchain(
+    name = "linux_arm_toolchain",
+    toolchain_identifier = "linux_arm-toolchain",
+    toolchain_config = ":linux_arm_toolchain_config",
+    all_files = ":empty",
+    compiler_files = ":empty",
+    dwp_files = ":empty",
+    linker_files = ":empty",
+    objcopy_files = ":empty",
+    strip_files = ":empty",
+    supports_param_files = 0,
+)
+
+toolchain(
+    name = "cc_toolchain_for_linux_arm",
+    toolchain = ":linux_arm_toolchain",
+    toolchain_type = "@bazel_tools//tools/cpp:toolchain_type",
+    exec_compatible_with = [
+        "@platforms//cpu:x86_64",
+        "@platforms//os:linux",
+    ],
+    target_compatible_with = [
+        "@platforms//cpu:armv7",
+        "@platforms//os:linux",
+    ],
+)
+
diff --git a/toolchain/cc_toolchain_config.bzl b/toolchain/cc_toolchain_config.bzl
new file mode 100644
index 0000000..32b67d3
--- /dev/null
+++ b/toolchain/cc_toolchain_config.bzl
@@ -0,0 +1,60 @@
+load("@bazel_tools//tools/cpp:cc_toolchain_config_lib.bzl", "tool_path")
+def _impl(ctx):
+    tool_paths = [ # NEW
+        tool_path(
+            name = "gcc",
+            path = "/usr/bin/aarch64-linux-gnu-gcc",
+        ),
+        tool_path(
+            name = "ld",
+            path = "/usr/bin/aarch64-linux-gnu-ld",
+        ),
+        tool_path(
+            name = "ar",
+            path = "/usr/bin/aarch64-linux-gnu-ar",
+        ),
+        tool_path(
+            name = "cpp",
+            path = "/usr/bin/aarch64-linux-gnu-cpp",
+        ),
+        tool_path(
+            name = "gcov",
+            path = "/usr/bin/aarch64-linux-gnu-gcov",
+        ),
+        tool_path(
+            name = "nm",
+            path = "/usr/bin/aarch64-linux-gnu-nm",
+        ),
+        tool_path(
+            name = "objdump",
+            path = "/usr/bin/aarch64-linux-gnu-objdump",
+        ),
+        tool_path(
+            name = "strip",
+            path = "/usr/bin/aarch64-linux-gnu-strip",
+        ),
+    ]
+
+    return cc_common.create_cc_toolchain_config_info(
+        ctx = ctx,
+        cxx_builtin_include_directories = [
+                "/usr/aarch64-linux-gnu/include",
+                "/usr/lib/gcc-cross/aarch64-linux-gnu/14/include",
+        ],
+        toolchain_identifier = "armv7-toolchain",
+        host_system_name = "local",
+        target_system_name = "local",
+        target_cpu = "armv7",
+        target_libc = "unknown",
+        compiler = "gcc",
+        abi_version = "gnueabi",
+        abi_libc_version = "unknown",
+        tool_paths = tool_paths, # NEW
+    )
+
+cc_toolchain_config = rule(
+    implementation = _impl,
+    attrs = {},
+    provides = [CcToolchainConfigInfo],
+)
+
